parsing/src/parts.ts

```diff
@@ -183,17 +204,21 @@ function getItems(data: any[]): PartDataInterface {
                     .find((entry: any) => entry.ClassName === `Desc_${productClass}_C`);
 
                 // Extract stack size
-                const stackSize = stackSizeConvert(classDescriptor?.mStackSize || "SS_UNKNOWN")
+                const stackSize: number = stackSizeConvert(classDescriptor?.mStackSize || "SS_UNKNOWN")
+                // Extract the energy value
+                const energyValue: number = classDescriptor?.mEnergyValue;
```

Comment:
```suggestion
                const energyValue: number = classDescriptor?.mEnergyValue ?? 0;
```

Null coalesce will always ensure this has a value of some number or zero, as right now it could be undefined. Amazed typescript let you do this tbh!

Reply:
If classDescriptor is null or undefined, the expression will short-circuit and return undefined instead of throwing an error. But: I like your suggestion better.